### PR TITLE
MOB-3650 call loyalty pre-auth in earn mode

### PR DESCRIPTION
**Pull request by dianakarh**
## JIRA
[MOB-3650](https://karhoo.atlassian.net/browse/MOB-3650)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift [hunk 1]
```diff
@@ -279,7 +285,9 @@ final class KarhooLoyaltyPresenter: LoyaltyPresenter {
             return
         }
         
-        let request = LoyaltyPreAuth(identifier: viewModel.loyaltyId, currency: viewModel.currency, points: viewModel.burnAmount, flexpay: false, membership: "")
+        let flexPay = currentMode == .earn // This will be adapted when we integrate flexPay
```
Comment (mohadian) [inline KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift:288]: We could add a function canFlexPay() and extract this logic

File: KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift [hunk 2]
```diff
@@ -279,7 +285,9 @@ final class KarhooLoyaltyPresenter: LoyaltyPresenter {
...
+        let points = currentMode == .burn ? viewModel.burnAmount : 0 // https://karhoo.atlassian.net/wiki/spaces/PAY/pages/4343201851/Loyalty
```
Comment (mohadian) [inline KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift:289]: We could add a function getLoyaltyBurnPoints and extract this logic

File: KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift (hunk 1 above)
Comment (dianakarh) [reply KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift]: done

File: KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift (hunk 2 above)
Comment (dianakarh) [reply KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift]: done

**Review by mohadian — APPROVED** [review]

**Pull request merged**